integration-cli/docker_cli_service_create_test.go

```diff
@@ -90,35 +91,128 @@ func (s *DockerSwarmSuite) TestServiceCreateWithSecretSimple(c *check.C) {
 	c.Assert(refs[0].File.Name, checker.Equals, testName)
 	c.Assert(refs[0].File.UID, checker.Equals, "0")
 	c.Assert(refs[0].File.GID, checker.Equals, "0")
+
+	out, err = d.Cmd("service", "rm", serviceName)
+	c.Assert(err, checker.IsNil, check.Commentf(out))
+	d.DeleteSecret(c, testName)
 }
 
-func (s *DockerSwarmSuite) TestServiceCreateWithSecretSourceTarget(c *check.C) {
+func (s *DockerSwarmSuite) TestServiceCreateWithSecretSourceTargetPaths(c *check.C) {
+	d := s.AddDaemon(c, true, true)
+
+	testPaths := map[string]string{
+		"app":             "/etc/secret",
+		"test_secret":     "test_secret",
+		"relative_secret": "relative/secret",
+		"invalid_secret":  "../secret",
```

Comment:
The way paths work is OS-specific, so deciding in the manager or the client whether a path is acceptable is a bit tricky. We don't know what OS the service will eventually run on.